Fix:quickstart.md add using Docker on UnionTech OS Desktop 20 Pro (#975)
apache/polaris: site/content/in-dev/unreleased/quickstart.md

Reviewer: Are these changes specific just to UOS? If so I would probably rather not change the quickstart itself; this is a lot of content for a setup that 99% of users will not encounter. We don't even have instructions for the most popular desktop OSs in the world :)

Reviewer 2: I guess it happens in certain Linux distributions, https://github.com/moby/moby/discussions/47297. I don't mind adding several lines in the doc to clarify it a bit.

Reviewer 3: Agreed on this. Quick google shows UOS is based off Deepin which is based off Debian and the ref ticket I shared earlier also pointed to Debian based OS. No reported issue for Mac/Windows/Ubuntu so far.

Reviewer: This just does not really seem in scope for the Polaris quickstart doc which we really need to keep quite lean. 

IIUC, it's teaching users of a niche OS how to run Docker images?

> No reported issue for Mac/Windows/Ubuntu so far.

We don't even try to tell users how to run any of this in a Windows-friendly way so I guess it makes sense there would be no reported issues

Author: You're absolutely correct. These modifications are tailored specifically for UnionTech OS (UOS). Although I can't be certain whether other users will face comparable challenges, I've already refined the updated content. The aim is to offer better support and guidance, enabling a broader user base to make seamless use of [the relevant product or service].

Reviewer: Yeah, I just think this is excessive. The `From Docker` section is currently **3 lines** and this takes it to **12**. By the end of the `From Docker Section`, some 12/25 of the lines in the **entire doc so far** are related to this fix. If it's important to a lot of users, maybe. But as things stand I think, at most, a one-line blurb that links to another doc is probably sufficient.

brew install --cask docker
```

> There could be a [Docker permission issues](https://github.com/apache/polaris/pull/971) related to seccomp configuration. To resolve these issues, set the `seccomp` profile to "unconfined" when running a container.

Reviewer 2: I think the code example is good to have, which give users a quick solution without digging into details.
```shell
docker run --security-opt seccomp=unconfined apache/polaris:latest
```

Author: The example has been updated.

brew install --cask docker
```

There could be a [Docker permission issues](https://github.com/apache/polaris/pull/971) related to seccomp configuration. To resolve these issues, set the `seccomp` profile to "unconfined" when running a container. For example:

```shell
docker run --security-opt seccomp=unconfined apache/polaris:latest

> 

Reviewer: You're missing a ``` here; the doc doesn't render properly

Author: Thanks for letting me know! I've fixed it and optimized the document anew.

brew install --cask docker
```

#### UnionTech OS Desktop 20 Pro

> When using Docker on UnionTech OS Desktop 20 Pro, you may encounter issues related to seccomp configuration. It is recommended to set the seccomp profile to "unconfined" to avoid potential permission problems.

Author: Modified to a more generic way as suggested, with sample trace information attached.

> When using Docker on UnionTech OS Desktop 20 Pro, you may encounter issues related to seccomp configuration. It is recommended to set the seccomp profile to "unconfined" to avoid potential permission problems.

Reviewer 3: @flyrain do u mind do a review for the wording?

> If you encounter Docker permission-related issues, you can refer to the link [gradle](https://github.com/apache/polaris/pull/971)
> 
> you may encounter issues related to seccomp configuration. It is recommended to set the seccomp profile to "unconfined" to avoid potential permission problems.
> 
> To set the seccomp profile to "unconfined" for a single container, you can use the following command when running the container

Reviewer 2: ```suggestion
There could be a [Docker permission issues](https://github.com/apache/polaris/pull/971) related to seccomp configuration. Set the seccomp profile to "unconfined" for a single container when running the container:
```

Reviewer: Oh, I see -- this can happen with any distro that uses seccmp. Have we tested this flow with e.g. CentOS?

> ```shell
> docker run --security-opt seccomp=unconfined <image_name>
> ```
> 
> Replace <image_name> with the actual Docker image you want to run, such as apache/polaris:latest.

Reviewer 2: ```suggestion
```shell
docker run --security-opt seccomp=unconfined apache/polaris:latest
```

Replace <image_name> with the actual Docker image you want to run, such as apache/polaris:latest.
```

Author: Thank you for the optimization! After the changes, it feels much more concise and intuitive.I've merged the optimized content.

> If you want to set the seccomp profile to "unconfined" globally for all Docker containers, you need to edit the Docker daemon configuration file. Open or create /etc/docker/daemon.json and add the following content

Reviewer 2: i'm not sure if we want to recommend this way here, this seems out of scope of the Polaris doc.

Author: I agree with you. According to your suggestion, the way indeed seems out of the scope of the Polaris doc. And I've already optimized this part of the content to make it more relevant.
